syntax = "proto3";

package myservice;

service MyService {
  // Unary RPC
  rpc UnaryHello (HelloRequest) returns (HelloResponse);

  // Server Streaming RPC
  rpc ServerStreamHello (HelloRequest) returns (stream HelloResponse);

  // Client Streaming RPC
  rpc ClientStreamHello (stream HelloRequest) returns (HelloResponse);

  // Bidirectional Streaming RPC
  rpc Chat (stream HelloRequest) returns (stream HelloResponse);
}

message HelloRequest {
  string name = 1;
  string message = 2; // Digunakan untuk chat
}

message HelloResponse {
  string message = 1;
}
